.github/workflows/build-and-test.yml

```diff
@@ -105,6 +107,7 @@ jobs:
         with:
           name: test-results
```

Comment:
i guess this needs to be something like `integration-test-results` as the pipeline complains that an artifact with that name already exists 🙈